cms/djangoapps/contentstore/views/tests/test_block.py

```diff
@@ -1106,6 +1107,57 @@ def test_duplicate_library_content_block(self):  # pylint: disable=too-many-stat
         assert dupe_html_2.display_name == "HTML 2 Title (Lib Update)"
         assert dupe_html_2.data == "HTML 2 Content (Lib Update)"
 
+    def test_duplicate_tags(self):
+        """
+        Test that duplicating a tagged XBlock also duplicates its content tags.
+        """
+        source_course = CourseFactory()
+        user = UserFactory.create()
+        source_chapter = BlockFactory(
+            parent=source_course, category="chapter", display_name="Source Chapter"
+        )
+        source_block = BlockFactory(parent=source_chapter, category="html", display_name="Child")
+
+        # Create a couple of taxonomies with tags
+        taxonomyA = tagging_api.create_taxonomy(name="A", export_id="A")
+        taxonomyB = tagging_api.create_taxonomy(name="B", export_id="B")
+        tagging_api.set_taxonomy_orgs(taxonomyA, all_orgs=True)
+        tagging_api.set_taxonomy_orgs(taxonomyB, all_orgs=True)
+        tagging_api.add_tag_to_taxonomy(taxonomyA, "one")
+        tagging_api.add_tag_to_taxonomy(taxonomyA, "two")
+        tagging_api.add_tag_to_taxonomy(taxonomyB, "three")
+        tagging_api.add_tag_to_taxonomy(taxonomyB, "four")
+
+        # Tag the chapter
+        tagging_api.tag_object(str(source_chapter.location), taxonomyA, ["one", "two"])
+        tagging_api.tag_object(str(source_chapter.location), taxonomyB, ["three", "four"])
+
+        # Tag the child block
+        tagging_api.tag_object(str(source_block.location), taxonomyA, ["two"],)
+
+        # Refresh.
+        source_chapter = self.store.get_item(source_chapter.location)
+        expected_chapter_tags = 'A:one,two;B:four,three'
+        assert source_chapter.serialize_tag_data(source_chapter.location) == expected_chapter_tags
```

Comment:
Can you just change this test so that it uses `tagging_api.get_object_tags` to get the tags, rather than `. serialize_tag_data()` ? 

Because `serialize_tag_data` assumes that we're using `TaggedBlockMixin` but that's an implementation detail that we may change in the near future, based on that discussion.

What I'd like to see is that even if we change to not include tags in the OLX, these tests will still be working and not need any changes.

Reply:
Done: 672f76cd7d8e54c0e6814be3616882d89857cb29